File: seminar_paper/small_experiment_buffer_overflow/language_specific_tasks/chat_gpt_response.py
def select_from_buffer(buffer):
    """
    Prompts user to select index from buffer and returns selected element.
    """
    print(f"Buffer has {len(buffer)} elements (0 to {len(buffer)-1})")
    
    while True:
        try:
            # Ask the user for input and convert it to an integer
            index = int(input(f"Please select an index (0 to {len(buffer)-1}): "))
            
            # Check if the index is within the valid range
            if 0 <= index < len(buffer):
                return buffer[index]
            else:
                print("Invalid index. Please select a number within the valid range.")
        
        except ValueError:
            print("Invalid input. Please enter a valid integer.")

# Example usage
buffer = ["apple", "banana", "cherry", "date"]
result = select_from_buffer(buffer)
print("Selected:", result)

File: seminar_paper/small_experiment_buffer_overflow/language_specific_tasks/chat_gpt_response.rs
use std::io;

fn select_element(buffer: &[i32]) -> Vec<i32> {
    let mut input = String::new();

    println!("Enter index:");

    // Read user input
    io::stdin().read_line(&mut input).expect("Failed to read line");

    // Parse the input to an integer
    let index: usize = match input.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Invalid input. Please enter a valid number.");
            return Vec::new(); // Return an empty vector if parsing fails
        }
    };

    // Check if the index is within bounds and return the selected element
    if index < buffer.len() {
        vec![buffer[index]]
    } else {
        println!("Index out of bounds.");
        Vec::new() // Return an empty vector if the index is out of bounds
    }
}

fn main() {
    let buffer = vec![10, 20, 30, 40, 50];

    let selected = select_element(&buffer);

    if !selected.is_empty() {
        println!("Selected element: {:?}", selected);
    } else {
        println!("No valid element selected.");
    }
}